### DISCUSSION TITLE
thread can not work on Firefox

### DISCUSSION THREAD

**Discussion by aisnote** [emscripten-core/emscripten#18179] [General]
my native c++ code may involve the thread like background processing. 
And I tried on Firefox, it stuck it.

On Chrome, it is ok.
I built with:
-s PTHREAD_POOL_SIZE=8 -s USE_PTHREADS=1 -s MAIN_MODULE=1 -pthread
And this is from the building:

em++: warning: -sMAIN_MODULE + pthreads is experimental [-Wexperimental]

Any ideas? thanks.

**Comment by sbc100** [comment]
You can try building with `-sPTHREADS_DEBUG`

**Reply by aisnote** [reply]
will try and back to here. Thanks.
